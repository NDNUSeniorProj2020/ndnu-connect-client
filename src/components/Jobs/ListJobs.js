import React from 'react';
import PropTypes from 'prop-types';

import JobCard from './JobCard';

export default function ListJobs({ jobs }) {
	return (
    <div>
      {jobs.map(job => (
        <div key={job.id}>
          <JobCard job={job} />
        </div>
      ))}
    </div>
  );
}

ListJobs.propTypes = { jobs: PropTypes.array };
ListJobs.defaultProps = { jobs: [] };
