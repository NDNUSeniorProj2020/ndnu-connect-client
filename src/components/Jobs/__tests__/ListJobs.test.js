import React from 'react';
import { shallow, mount } from 'enzyme';

import ListJobs from '../ListJobs';

describe('tests for ListJobs', () => {
	describe('snapshot tests', () => {
		it('renders without crashing without props', () => {
			const tree = shallow(<ListJobs />);
			expect(tree).toMatchSnapshot();
		});

		it('renders without crashing when props are passed', () => {
			const tree = shallow(<ListJobs jobs={[{ id: 1, title: 'Bus Driver' }]} />);
			expect(tree).toMatchSnapshot();
		});
	});

	describe('unit tests', () => {
		it('renders with default jobs prop if jobs prop is empty', () => {
			const wrapper = mount(<ListJobs />);
			expect(wrapper.props().jobs).toEqual([]);
		});
	});
});
