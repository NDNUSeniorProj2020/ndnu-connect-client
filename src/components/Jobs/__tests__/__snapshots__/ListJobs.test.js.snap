// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ListJobs snapshot tests renders without crashing when props are passed 1`] = `ShallowWrapper {}`;

exports[`tests for ListJobs snapshot tests renders without crashing without props 1`] = `ShallowWrapper {}`;
